service/raft/raft_group_registry.cc

```diff
@@ -538,8 +539,9 @@ future<bool> raft_server_with_timeouts::trigger_snapshot(seastar::abort_source*
 
 future<> raft_server_with_timeouts::read_barrier(seastar::abort_source* as, std::optional<raft_timeout> timeout)
 {
-    return run_with_timeout([&](abort_source* as) {
-        return _group_server.server->read_barrier(as);
+    return run_with_timeout([&](abort_source* as) -> future<> {
+        co_await utils::get_local_injector().inject("sleep_in_read_barrier", std::chrono::seconds(1));
```

Comment:
question: Should we extend the timeout to a bit longer? Slower machines processing a lot of workload may not be able to shut down Raft within a second, or so I imagine at least.